datafusion/execution/src/memory_pool/pool.rs

```diff
@@ -231,6 +235,10 @@ impl MemoryPool for FairSpillPool {
     }
 }
 
+/// Constructs a resources error based upon the individual [`MemoryReservation`].
+///
+/// The error references the total bytes affiliated with the reservation's
+/// [`MemoryConsumer`], and not the total within the collective [`MemoryPool`].
 #[inline(always)]
 fn insufficient_capacity_err(
```

Comment:
In a follow up PR, I would like to iterate on this error message.

Take the original/current:
`Failed to allocate additional {} bytes for {} with {} bytes already allocated - maximum available is {}`

And change into something like:
`Failed to allocate additional {} bytes for {} with {} bytes already allocated for this reservation - maximum available is {} for the total pool`